cl/assets/react/DisclosureViewer.tsx

```diff
@@ -0,0 +1,463 @@
+    fetchJudge(event);
+  };
+
+  const debounceFetchJudge = React.useMemo(() => debounce(changeHandler, 300), []);
+  const urlList = window.location.pathname.split('/');
+  const judgeUrl = [urlList[0], urlList[1], urlList[2], urlList[5], urlList[6]].join('/');
+  return (
+    <div>
+      {data != '' && data.has_been_extracted ? (
+        <div>
+          <div className={'v-offset-below-3 v-offset-above-3'}>
+            <div className={'col-lg-9'}>
+              {Tabs(data, years, years[indx], fetchDisclosure, doc_ids, judge_name, judgeUrl)}
+              <div className="tabcontent">
+                {TableMaker(data, 'investments', is_admin)}
+                {TableMaker(data, 'gifts', is_admin)}
+                {TableMaker(data, 'reimbursements', is_admin)}
+                {TableMaker(data, 'spouse_incomes', is_admin)}
+                {TableMaker(data, 'debts', is_admin)}
+                {TableMaker(data, 'non_investment_incomes', is_admin)}
+                {TableMaker(data, 'agreements', is_admin)}
+                {TableMaker(data, 'positions', is_admin)}
+                {data.addendum_content_raw != '' ? (
+                  <>
+                    <h3>Addendum</h3>
+                    <article>{data.addendum_content_raw}</article>{' '}
+                  </>
+                ) : (
+                  ''
+                )}
+              </div>
+            </div>
+            <div className={'col-lg-3'}>{Sidebar(data, is_admin, judge, debounceFetchJudge, visible, setVisible)}</div>
+          </div>
+        </div>
+      ) : data != '' && data.has_been_extracted == false ? (
+        <div>
+          <div className={'v-offset-below-3 v-offset-above-3'}>
+            <div className={'col-lg-9'}>
+              {Tabs(data, years, years[indx], fetchDisclosure, doc_ids, judge_name, judgeUrl)}
+              {/*{Tabs(data, years, year, fetchDisclosure, doc_ids, judge_name)}*/}
+              <div className="tabcontent">
+                <div className={'text-center v-offset-above-4'}>
+                  <i className="fa fa-exclamation-triangle gray" />
+                  <h1>Table extraction failed.</h1>
+                  <p>You can still view this Financial Disclosure by clicking the thumbnail.</p>
+                </div>
+              </div>
+            </div>
+            <div className={'col-lg-3'}>{Sidebar(data, is_admin, judge, debounceFetchJudge, visible, setVisible)}</div>
+          </div>
+        </div>
+      ) : (
+        <h1 className={'text-center'}>Loading...</h1>
+      )}
+    </div>
+  );
+};
+
+const TableMaker = (data: Data, key: string, is_admin: boolean) => {
+  const url = data.filepath;
+  const disclosure_id = data.id;
+  const rows = data[key];
+  const fields = disclosureModel[key]['fields'];
+  const title: string = disclosureModel[key]['title'];
+  let api_key = key.replaceAll('_', '-');
+  if (api_key == 'positions') {
+    api_key = `disclosure-${api_key}`;
+  }
```

Comment:
Why do this instead of getting the values you want above? E.g., include dashes not underscores up there, and use disclosure-positions instead of positions?

Reply:
see #1781

Reply:
I just mean, up above you define the API endpoints as things like `spouse_income` when you could call them `spouse-income` with the dash already in it. And instead of calling it `positions` up above, couldn't you call it `disclosure-positions`?